src/waveform/renderers/waveformrenderbeat.cpp

```diff
@@ -76,9 +76,11 @@ void WaveformRenderBeat::draw(QPainter* painter, QPaintEvent* /*event*/) {
     const float rendererHeight = m_waveformRenderer->getHeight();
 
     int beatCount = 0;
+    QMap<Beat, float> beatPositions;
 
     while (it->hasNext()) {
-        double beatPosition = it->next();
+        auto beat = it->next();
+        double beatPosition = beat.getSamplePosition();
         double xBeatPoint =
                 m_waveformRenderer->transformSamplePositionInRendererWorld(beatPosition);
```

Comment:
Can we come up with a better description than "x"? What is "x"? A position within the widget? What are the units? Pixels?